### PR TITLE
Week_2_Task

### PR DESCRIPTION
**Pull request by ssinha2103**

### CODE REVIEW

**Comment by anirvan-majumdar** [comment]
## Quick observations:
1. Do not raise PRs without a clear description of tasks.
2. Do not raise PRs that have so many changes. Identify blocks of changes and raise separate PRs for each.
3. Setup separate branches for each block/module of tasks and raise separate PRs from them.
4. You need to have better comments for your commits. Each commit statement should provide a short summary of what it is about.
5. Write pydocs!
6. Learn how to write pydocs properly ([reference](https://peps.python.org/pep-0257/))

File: UP_onboarding_system/point_of_sale/admin.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from .models import Items, Orders, Profile, Stores
+from django.contrib import admin
+
+admin.site.register([Profile, Stores, Items, Orders])
```
Comment (aravinda-s-holla) [inline UP_onboarding_system/point_of_sale/admin.py:4]: Add new line

File: UP_onboarding_system/point_of_sale/serializers.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+class ProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Profile
+        fields = ("name", "role",)
+
+
+class UserRegistrationSerializer(serializers.ModelSerializer):
+    profile = ProfileSerializer()
+    password = serializers.CharField(style={'input_type': 'password'}, write_only=True)
```
Comment (aravinda-s-holla) [inline UP_onboarding_system/point_of_sale/serializers.py:16]: Can take password2  and  reassert that is  password1

File: UP_onboarding_system/point_of_sale/serializers.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        return lower_email
+
+    def create(self, validated_data):
+        profile_data = validated_data.pop("profile")
+        user = User.objects.create(
+            username=validated_data["username"],
+            email=validated_data["email"],
+        )
+        user.set_password(validated_data["password"])
+        user.save()
```
Comment (aravinda-s-holla) [inline UP_onboarding_system/point_of_sale/serializers.py:40]: Serializer can itself create object. Refer: https://www.django-rest-framework.org/api-guide/serializers/

File: UP_onboarding_system/point_of_sale/serializers.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+class UserViewSerializers(serializers.ModelSerializer):
+    profile = ProfileSerializer()
+
+    class Meta:
+        model = User
+        fields = (
+            "username",
+            "email",
+            "profile",
+        )
```
Comment (aravinda-s-holla) [inline UP_onboarding_system/point_of_sale/serializers.py:116]: This can fit in one line

File: UP_onboarding_system/point_of_sale/tasks.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+@shared_task()
+def create_an_order(pk, data):
+    #import ipdb; ipdb.set_trace()
+    # print("CELERY SSSSSSSSSSSSSSSSSS")
+    #_logger.info("CELERY Start")
+    # sleep(5)
+    serializer = OrderSerializer(data=data)
+    user = User.objects.get(pk=pk)
+    serializer.is_valid()
```
Comment (aravinda-s-holla) [inline UP_onboarding_system/point_of_sale/tasks.py:23]: What if serializer has errors?

File: UP_onboarding_system/point_of_sale/tests/conftest.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+from point_of_sale.models import Profile
+
+User = get_user_model()
+
+
+@pytest.fixture
+def merchant_data():
+    merchant = User.objects.create_user(username="merchant", email="merchant@gmail.com", password="password@123")
+    profile = Profile.objects.create(user=merchant, name="Merchant_User", role=1)
+    return profile
```
Comment (aravinda-s-holla) [inline UP_onboarding_system/point_of_sale/tests/conftest.py:12]: can use random value using https://faker.readthedocs.io/

File: UP_onboarding_system/point_of_sale/admin.py (hunk 1 above)
Comment (ssinha2103) [reply UP_onboarding_system/point_of_sale/admin.py:4]: Added

File: UP_onboarding_system/point_of_sale/serializers.py (hunk 2 above)
Comment (ssinha2103) [reply UP_onboarding_system/point_of_sale/serializers.py:16]: Done

File: UP_onboarding_system/point_of_sale/serializers.py (hunk 4 above)
Comment (ssinha2103) [reply UP_onboarding_system/point_of_sale/serializers.py:116]: fitted in 1 line

File: UP_onboarding_system/point_of_sale/tasks.py (hunk 5 above)
Comment (ssinha2103) [reply UP_onboarding_system/point_of_sale/tasks.py:23]: added checks and validation errors in respective places

**Pull request merged**